### PR TITLE
46684

### PR DESCRIPTION
**Pull request by rth-axelor**

### CODE REVIEW

File: axelor-account/src/main/java/com/axelor/apps/account/web/AccountController.java [hunk 1]
```diff
@@ -123,12 +125,34 @@ public void manageAnalytic(ActionRequest request, ActionResponse response) {
...
+        specificAnalyticDistributionTemplate =
+            Beans.get(AnalyticDistributionTemplateService.class)
+                .personalizeAnalyticDistributionTemplate(
+                    analyticDistributionTemplate, account.getCompany());
+      }
+      if ((analyticDistributionTemplate == null || !analyticDistributionTemplate.getIsSpecific())
+          && specificAnalyticDistributionTemplate != null) {
+        response.setValue("analyticDistributionTemplate", specificAnalyticDistributionTemplate);
+        response.setView(
+            ActionView.define("Specific Analytic Distribution Template")
```
Comment (ale-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/web/AccountController.java:146]: Please translate this title

**Review by ale-axelor — CHANGES_REQUESTED** [review]

File: axelor-account/src/main/java/com/axelor/apps/account/web/AccountController.java [hunk 2]
```diff
@@ -123,14 +125,35 @@ public void manageAnalytic(ActionRequest request, ActionResponse response) {
...
+      AnalyticDistributionTemplate analyticDistributionTemplate =
+          account.getAnalyticDistributionTemplate();
+      if (analyticDistributionTemplate == null && account.getAnalyticDistributionAuthorized()) {
+        specificAnalyticDistributionTemplate =
             Beans.get(AnalyticDistributionTemplateService.class)
                 .createDistributionTemplateFromAccount(account);
-        if (analyticDistributionTemplate != null) {
-          response.setValue("analyticDistributionTemplate", analyticDistributionTemplate);
-        }
+        System.err.println(specificAnalyticDistributionTemplate);
```
Comment (ale-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/web/AccountController.java:135]: Please remove this System.err.println

**Review by ale-axelor — CHANGES_REQUESTED** [review]

File: axelor-account/src/main/java/com/axelor/apps/account/service/analytic/AnalyticDistributionTemplateServiceImpl.java [hunk 3]
```diff
@@ -169,6 +169,8 @@ protected void printCheckAnalyticDistributionTemplateCompany(
     }
   }
 
+  @Override
+  @Transactional
```
Comment (ale-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/service/analytic/AnalyticDistributionTemplateServiceImpl.java:173]: Please see our documentation/convention with Transactional annotation

File: axelor-account/src/main/java/com/axelor/apps/account/web/AccountController.java [hunk 4]
```diff
@@ -123,14 +125,34 @@ public void manageAnalytic(ActionRequest request, ActionResponse response) {
...
-        }
+      } else if (account.getAnalyticDistributionAuthorized()
+          && analyticDistributionTemplate != null) {
+        specificAnalyticDistributionTemplate =
+            Beans.get(AnalyticDistributionTemplateService.class)
+                .personalizeAnalyticDistributionTemplate(
+                    analyticDistributionTemplate, account.getCompany());
+      }
+      if ((analyticDistributionTemplate == null || !analyticDistributionTemplate.getIsSpecific())
+          && specificAnalyticDistributionTemplate != null) {
```
Comment (ale-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/web/AccountController.java:143]: Please move this logic to a new method in a service

File: axelor-account/src/main/java/com/axelor/apps/account/web/FixedAssetController.java [hunk 5]
```diff
@@ -230,7 +230,7 @@ public void personalizeAnalyticDistributionTemplate(
       if (analyticDistributionTemplate == null || !analyticDistributionTemplate.getIsSpecific()) {
         response.setValue("analyticDistributionTemplate", specificAnalyticDistributionTemplate);
         response.setView(
-            ActionView.define("Specific Analytic Distribution Template")
+            ActionView.define(I18n.get("Specific Analytic Distribution Template"))
```
Comment (ale-axelor) [inline axelor-account/src/main/java/com/axelor/apps/account/web/FixedAssetController.java:233]: Please create a constant in IExceptionMessage

**Review by ale-axelor — CHANGES_REQUESTED** [review]

**Review by jmi-axelor — APPROVED** [review]

**Pull request merged**